src/data-table/data-table.tsx

```diff
@@ -180,17 +188,25 @@ export const DataTable = ({
                 data-testid={getDataTestId(`row-${cellId}`)}
                 key={cellId}
                 style={style}
+                $isCheckbox={false}
             >
-                {cellData}
+                {typeof cellData === "string" ? (
+                    <BodyCellContent>{cellData}</BodyCellContent>
+                ) : (
+                    cellData
```

Comment:
something that I missed out, if `cellData` is a function, it should be called with the row item and `RowRenderProps`